Lares/Assets/Player/Scripts/PlayerController.cs

```diff
@@ -0,0 +1,202 @@
+            GameObject model = transform.GetChild(1).gameObject;
+
+            while (_moveDirection != Vector3.zero)
+            {
+                while (_isMovementLocked)
+                {
+                    yield return new WaitForFixedUpdate();
+                }
+
+                rotPoint.transform.rotation = Quaternion.Slerp(rotPoint.transform.rotation,
+                    Quaternion.Euler(0, _cameraTransform.rotation.eulerAngles.y, 0), _rotationSpeed);
+
+                if (OnIncline())
+                {
+                    _rigidbody.AddForce(GetInclineForwardDirection() * 20, ForceMode.Force);
+                }
+
+                float currentMovementForce = _movementForce;
+                float currentMaxSpeed = _maxSpeed;
+                if (_isSprinting) { currentMovementForce *= _sprintSpeedMultiplier; currentMaxSpeed *= _sprintSpeedMultiplier; }
+
+                _rigidbody.AddForce(rotPoint.transform.forward * (_moveDirection.z * currentMovementForce * Time.fixedDeltaTime),
+                    ForceMode.VelocityChange);
+                _rigidbody.AddForce(rotPoint.transform.right * (_moveDirection.x * currentMovementForce * Time.fixedDeltaTime),
+                    ForceMode.VelocityChange);
+
+                model.transform.rotation = Quaternion.Slerp(model.transform.rotation,
+                    Quaternion.LookRotation(new Vector3(_rigidbody.linearVelocity.x, 0f, _rigidbody.linearVelocity.z)),
+                    _rotationSpeed);
+
+                _rigidbody.linearVelocity = Vector3.ClampMagnitude(_rigidbody.linearVelocity, currentMaxSpeed);
+
+
+                _rigidbody.useGravity = !OnIncline();
+                yield return new WaitForFixedUpdate();
+            }
+            _moveCoroutine = null;
+        }
+
+        private void Jump(InputAction.CallbackContext context)
+        {
+            if (!_canJump) { return; }
+            _rigidbody.AddForce(new Vector3(0f, _jumpForce - _rigidbody.linearVelocity.y, 0f), ForceMode.Impulse);
+            _jumpCount++;
+
+        }
+
+        private bool OnIncline()
+        {
+            if (Physics.Raycast(transform.position, Vector3.down, out _inclineHitOut, _maxRaycastHitDistance))
+            {
+                float angle = Vector3.Angle(Vector3.up, _inclineHitOut.normal);
+                bool angleAboveInclineRange = angle < _maxInclineAngle && angle != 0.0f;
+                return angleAboveInclineRange;
+            }
+            return false;
+        }
+
+        private Vector3 GetInclineForwardDirection()
+        {
+            return Vector3.ProjectOnPlane(_rigidbody.linearVelocity, _inclineHitOut.normal);
+        }
+
+        #region Collisions Checks
+        private void OnCollisionEnter(Collision collision)
+        {
+            if (collision.gameObject.CompareTag("Ground"))
+            {
+                groundColliderCount++;
+                _jumpCount = 0;
+                _isGrounded = true;
+            }
```

Comment:
I don't know if this is simply temporary (would be nice to comment that it is, if so), however, this is prone to error.

For example: if you stood on a wall, you would not be marked as touching the ground - maybe worth swapping this to a very short raycast?